app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+
+  def new
+  end
+
+  def create
+    customer = Customer.find_by_email(params[:email])
+    if customer == customer.authenticate(params[:password])
```

Comment:
I think I typoed 
 == should be &&
if customer(*found by Customer.find...*) && customer.authenticate(params[:password]) are true then you get your session and redirected, else back to signin.....

or do I even need the authenticate line because I cut out the code and left just ' if customer ' 
and am still able to login...